File: rest-service/src/main/java/org/veupathdb/service/multiblast/model/blast/TBlastnConfig.java
package org.veupathdb.service.multiblast.model.blast;

import java.io.File;

public class TBlastnConfig extends StdBlastConfig
{
  private TBlastnTask          task;
  private Integer              gapOpen;
  private Integer              gapExtend;
  private Double               xDropGap;
  private Double               xDropGapFinal;
  private Boolean              sumStats;
  private Integer              dbGenCode;
  private Boolean              ungapped;
  private Integer              maxIntronLength;
  private Seg                  seg;
  private TBlastnScoringMatrix matrix;
  private Double               threshold;
  private Integer              cullingLimit;
  private Double               bestHitOverhang;
  private Double               bestHitScoreEdge;
  private Boolean              subjectBestHit;
  private CompBasedStats       compBasedStats;
  private Boolean              useSmithWatermanAlignments;
  private File                 inPssm;

  public TBlastnTask getTask() {
    return task;
  }

  public TBlastnConfig setTask(TBlastnTask task) {
    this.task = task;
    return this;
  }

  public Integer getGapOpen() {
    return gapOpen;
  }

  public TBlastnConfig setGapOpen(Integer gapOpen) {
    this.gapOpen = gapOpen;
    return this;
  }

  public Integer getGapExtend() {
    return gapExtend;
  }

  public TBlastnConfig setGapExtend(Integer gapExtend) {
    this.gapExtend = gapExtend;
    return this;
  }

  public Double getxDropGap() {
    return xDropGap;
  }

  public TBlastnConfig setxDropGap(Double xDropGap) {
    this.xDropGap = xDropGap;
    return this;
  }

  public Double getxDropGapFinal() {
    return xDropGapFinal;
  }

  public TBlastnConfig setxDropGapFinal(Double xDropGapFinal) {
    this.xDropGapFinal = xDropGapFinal;
    return this;
  }

  public Boolean getSumStats() {
    return sumStats;
  }

  public TBlastnConfig setSumStats(Boolean sumStats) {
    this.sumStats = sumStats;
    return this;
  }

  public Integer getDbGenCode() {
    return dbGenCode;
  }

  public TBlastnConfig setDbGenCode(Integer dbGenCode) {
    this.dbGenCode = dbGenCode;
    return this;
  }

  public Boolean getUngapped() {
    return ungapped;
  }

  public TBlastnConfig setUngapped(Boolean ungapped) {
    this.ungapped = ungapped;
    return this;
  }

  public Integer getMaxIntronLength() {
    return maxIntronLength;
  }

  public TBlastnConfig setMaxIntronLength(Integer maxIntronLength) {
    this.maxIntronLength = maxIntronLength;
    return this;
  }

  public Seg getSeg() {
    return seg;
  }

  public TBlastnConfig setSeg(Seg seg) {
    this.seg = seg;
    return this;
  }

  public TBlastnScoringMatrix getMatrix() {
    return matrix;
  }

  public TBlastnConfig setMatrix(TBlastnScoringMatrix matrix) {
    this.matrix = matrix;
    return this;
  }

  public Double getThreshold() {
    return threshold;
  }

  public TBlastnConfig setThreshold(Double threshold) {
    this.threshold = threshold;
    return this;
  }

  public Integer getCullingLimit() {
    return cullingLimit;
  }

  public TBlastnConfig setCullingLimit(Integer cullingLimit) {
    this.cullingLimit = cullingLimit;
    return this;
  }

  public Double getBestHitOverhang() {
    return bestHitOverhang;
  }

  public TBlastnConfig setBestHitOverhang(Double bestHitOverhang) {
    this.bestHitOverhang = bestHitOverhang;
    return this;
  }

  public Double getBestHitScoreEdge() {
    return bestHitScoreEdge;
  }

  public TBlastnConfig setBestHitScoreEdge(Double bestHitScoreEdge) {
    this.bestHitScoreEdge = bestHitScoreEdge;
    return this;
  }

  public Boolean getSubjectBestHit() {
    return subjectBestHit;
  }

  public TBlastnConfig setSubjectBestHit(Boolean subjectBestHit) {
    this.subjectBestHit = subjectBestHit;
    return this;
  }

  public CompBasedStats getCompBasedStats() {
    return compBasedStats;
  }

  public TBlastnConfig setCompBasedStats(CompBasedStats compBasedStats) {
    this.compBasedStats = compBasedStats;
    return this;
  }

  public Boolean getUseSmithWatermanAlignments() {
    return useSmithWatermanAlignments;
  }

  public TBlastnConfig setUseSmithWatermanAlignments(Boolean useSmithWatermanAlignments) {
    this.useSmithWatermanAlignments = useSmithWatermanAlignments;
    return this;
  }

  public File getInPssm() {
    return inPssm;
  }

  public TBlastnConfig setInPssm(File inPssm) {
    this.inPssm = inPssm;
    return this;
  }
}

File: lib-jvm-mblast-sdk/src/main/kotlin/org/veupathdb/lib/mblast/sdk/query/blast/TBlastNConfig.kt
package org.veupathdb.lib.mblast.sdk.query.blast

import com.fasterxml.jackson.annotation.JsonGetter
import com.fasterxml.jackson.annotation.JsonSetter

class TBlastNConfig : BlastQueryConfig() {

  override val tool: BlastQueryTool
    get() = BlastQueryTool.TBlastN

  @get:JsonGetter("task")
  @set:JsonSetter("task")
  var task: TBlastNTask? = null

  @get:JsonGetter("wordSize")
  @set:JsonSetter("wordSize")
  var wordSize: UInt? = null

  @get:JsonGetter("gapOpen")
  @set:JsonSetter("gapOpen")
  var gapOpen: Int? = null

  @get:JsonGetter("gapExtend")
  @set:JsonSetter("gapExtend")
  var gapExtend: Int? = null

  @get:JsonGetter("dbGenCode")
  @set:JsonSetter("dbGenCode")
  var dbGenCode: Int? = null

  @get:JsonGetter("maxIntronLength")
  @set:JsonSetter("maxIntronLength")
  var maxIntronLength: UInt? = null

  @get:JsonGetter("matrix")
  @set:JsonSetter("matrix")
  var matrix: TBlastNMatrix? = null

  @get:JsonGetter("threshold")
  @set:JsonSetter("threshold")
  var threshold: Double? = null

  @get:JsonGetter("compBasedStats")
  @set:JsonSetter("compBasedStats")
  var compBasedStats: TBlastNCompBasedStats? = null

  @get:JsonGetter("seg")
  @set:JsonSetter("seg")
  var seg: BlastSeg? = null

  @get:JsonGetter("dbSoftMask")
  @set:JsonSetter("dbSoftMask")
  var dbSoftMask: String? = null

  @get:JsonGetter("dbHardMask")
  @set:JsonSetter("dbHardMask")
  var dbHardMask: String? = null

  @get:JsonGetter("cullingLimit")
  @set:JsonSetter("cullingLimit")
  var cullingLimit: UInt? = null

  @get:JsonGetter("bestHitOverhang")
  @set:JsonSetter("bestHitOverhang")
  var bestHitOverhang: Double? = null

  @get:JsonGetter("bestHitScoreEdge")
  @set:JsonSetter("bestHitScoreEdge")
  var bestHitScoreEdge: Double? = null

  @get:JsonGetter("subjectBestHit")
  @set:JsonSetter("subjectBestHit")
  var subjectBestHit: Boolean? = null

  @get:JsonGetter("sumStats")
  @set:JsonSetter("sumStats")
  var sumStats: Boolean? = null

  @get:JsonGetter("xDropoffPrelimGapped")
  @set:JsonSetter("xDropoffPrelimGapped")
  var xDropoffPrelimGapped: Double? = null

  @get:JsonGetter("xDropoffFinalGapped")
  @set:JsonSetter("xDropoffFinalGapped")
  var xDropoffFinalGapped: Double? = null

  @get:JsonGetter("ungappedOnly")
  @set:JsonSetter("ungappedOnly")
  var ungappedOnly: Boolean? = null

  @get:JsonGetter("useSWTraceback")
  @set:JsonSetter("useSWTraceback")
  var useSWTraceback: Boolean? = null
}